packages/components-library/src/components/ui/Pagination.vue

```diff
@@ -0,0 +1,287 @@
+<template>
+<nav class="c-pagination">
+
+  <div class="button-container">
+
+    <button v-if="val.count > val.size" :class="styles" class="t-page-prev btn btn-outline-primary mr-4"
+      :disabled="val.page <= 1"
+      @click="navigate(val.page - 1)">
+        &laquo;
+    </button>
+
+    <div class="btn-group">
+      <template v-if="ellipsisFirst">
+        <button :class="styles" class="btn btn-outline-primary" @click="navigate(ellipsisFirst)">
+            {{ellipsisFirst}}
+        </button>
+        <button :class="styles" class="btn ellipsis" disabled>...</button>
+      </template>
+
+      <button
+        class="btn btn-outline-primary"
+        :class="{'active': pageNumber === val.page, ...styles}" v-for="pageNumber in pages" :key="pageNumber"
+        @click="navigate(pageNumber)">
+        {{pageNumber}}
+      </button>
+
+      <template v-if="ellipsisLast">
+        <button :class="styles" class="btn ellipsis" disabled>...</button>
+        <button :class="styles" class="btn btn-outline-primary" @click="navigate(ellipsisLast)">
+          {{ellipsisLast}}
+        </button>
+      </template>
+
+      <button v-if="val.count > val.size"
+        :class="styles" class="t-page-next btn btn-outline-primary ml-4"
+        :disabled="val.page >= pageCount"
+        @click="navigate(val.page + 1)">
+        &raquo;
+      </button>
+    </div>
+
+    <form v-if="val.count > val.size" class="pagination-controls form-inline ml-2">
+      <div class="btn-group">
+        <select  class="form-control-sm ml-2 mr-2" v-model="val.size">
+          <option value="10">10</option>
+          <option value="20">20 </option>
+          <option value="50">50</option>
+        </select>
+        <label for="exampleInputEmail1">per page</label>
+      </div>
+    </form>
+  </div>
+
+  <div class="item-count mt-2">{{val.count}} entities total</div>
+</nav>
+</template>
+
+<script>
+export default {
+  async created () {
+    this.val = { ...this.val, ...this.value }
+    // Life-cycle hook for router-less pagination.
+    if (!this.useRouter) {
+      // Allow Vue instance listeners to initialize in tests.
+      await this.$nextTick()
+      // Pagination state is leading for the initial page,
+      // in case no router is being use.
+      this.fetchData(this.val.page)
+    }
+  },
+  computed: {
+    ellipsisFirst () {
+      const firstPage = this.pages[0]
+      if (firstPage > 2) { return 1 }
+      return null
+    },
+    ellipsisLast () {
+      const lastPage = this.pages[this.pages.length - 1]
+      if (lastPage <= (this.pageCount - 2)) { return this.pageCount }
+      return null
+    },
+    pageCount () {
+      if (!this.val.count) { return 0 }
+      return Math.floor(parseInt(this.val.count, 10) / this.val.size)
+    },
+    pages () {
+      const pages = []
+
+      const midPage = Math.floor(this.midPageSize / 2)
+      const isMidRange = this.val.page > midPage && !(this.pageCount < this.relativePageSize)
```

Comment:
https://sonarcloud.io/project/issues?id=molgenis_molgenis-frontend&pullRequest=464&resolved=false&types=CODE_SMELL

Reply:
Remove the condition; it wasn't used properly at all